designer/server/src/main/scala/pl/touk/nussknacker/ui/api/NodesApiHttpService.scala

```diff
@@ -144,6 +148,39 @@ class NodesApiHttpService(
       }
   }
 
+  expose {
+    nodesApiEndpoints.fetchLatestRecordsForNodeEndpoint
+      .serverSecurityLogic(authorizeKnownUser[NodesError])
+      .serverLogicEitherT { implicit loggedUser =>
+        { case (scenarioName, numberOfRecords, fetchLatestRecordsDto) =>
```

Comment:
I'm guessing that we don't validate if `numberOfRecords` is too big because we assume that `scenarioTestService` already do this because there is a limit for data generation for scenario testing purpose?

It is not visible in the code. If we `scenarioTestService` returned a business errors, we would see it. As an alternative, we can at least add a comment about that assumption.

Reply:
I added such comment to `NodesApiHttpService` 

              // The service is expected to limit the number of returned records if the response is too large.
              // If it does not, we may need to implement client-side pagination or request size limits.

Reply:
I dug a little bit more into the code and found out where this records count is done for test data generation mechanism, take a look at this:
```
      _ <- Either.cond(
        testSampleSize <= testDataSettings.maxSamplesCount,
        (),
        s"Too many samples requested, limit is ${testDataSettings.maxSamplesCount}"
      )
```
in `ScenarioTestService`. At the beginning, I thought that it is somewhere deeper and your endpoint do this check implicitly, but it's not. We have to reuse this code, IMO it is not acceptable that the API accepts any limit. User might blow up our heap memory.